core/src/appleMain/kotlin/Bluetooth.kt

```diff
@@ -0,0 +1,38 @@
+package com.juul.kable
+
+import com.juul.kable.Bluetooth.Availability.Available
+import com.juul.kable.Bluetooth.Availability.Unavailable
+import com.juul.kable.Reason.Resetting
+import com.juul.kable.Reason.Unauthorized
+import com.juul.kable.Reason.Unknown
+import com.juul.kable.Reason.Unsupported
+import kotlinx.cinterop.UnsafeNumber
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import platform.CoreBluetooth.CBCentralManagerStatePoweredOff
+import platform.CoreBluetooth.CBCentralManagerStatePoweredOn
+import platform.CoreBluetooth.CBCentralManagerStateResetting
+import platform.CoreBluetooth.CBCentralManagerStateUnauthorized
+import platform.CoreBluetooth.CBCentralManagerStateUnsupported
+
+/** https://developer.apple.com/documentation/corebluetooth/cbmanagerstate */
+public actual enum class Reason {
+    Off, // CBManagerState.poweredOff
+    Resetting, // CBManagerState.resetting
+    Unauthorized, // CBManagerState.unauthorized
+    Unsupported, // CBManagerState.unsupported
+    Unknown, // CBManagerState.unknown
+}
+
+@OptIn(UnsafeNumber::class)
+internal actual val bluetoothAvailability: Flow<Bluetooth.Availability> =
+    CentralManager.Default.delegate.state.map { state ->
+        when (state) {
+            CBCentralManagerStatePoweredOn -> Available
+            CBCentralManagerStatePoweredOff -> Unavailable(reason = Unknown)
```

Comment:
IMHO would be nice to surface `PoweredOff` as a state rather than `Unknown` for situations where the user has enabled Airplane mode so the UI can present advice accordingly.

Reply:
Unfortunately JS (https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/onavailabilitychanged) only provides a boolean of availability, so it degrades the common API to the same available/unavailable state. 🤷

Reply:
Good call out. 👍